lib/shelf_io.dart

```diff
@@ -74,8 +74,14 @@ Future handleRequest(HttpRequest request, Handler handler) async {
   try {
     shelfRequest = _fromHttpRequest(request);
   } catch (error, stackTrace) {
-    var response =
-        _logTopLevelError('Error parsing request.\n$error', stackTrace);
+    final isClientError = error is ArgumentError &&
+        (error.name == 'method' || error.name == 'requestedUri');
+    _logTopLevelError('Error parsing request.\n$error', stackTrace);
```

Comment:
We shouldn't log a top-level error if this was a client error.

Reply:
If the package would use `package:logging` we could use a different log level, but with this one, I think it is better to log top level then omit it.